### PR TITLE
Add PR template

### PR DESCRIPTION
**Pull request by folmos-at-orange**

### CODE REVIEW

**Review by tramora — APPROVED** [review]
Looks good for me

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+assignees: ''
+
+---
+
+_Put your message here_
+
+---
+
+### TODO
+- [ ] Self-Review: Review "Files Changed" tab and fix any problems you find before requesting
```
Comment (popescu-v) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:15]: I would add one more item:
```md
- [  ] Self-review: Make sure all PR checks (GitHub workflows) run and finish successfully before requesting a review
```

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+---
+name: Generic PR template
```
Comment (popescu-v) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:2]: I'm not sure we need to follow this template for `dev` -> `main` PRs on pending releases: it seems to me that the checklists should be simplified for this case (as already checked in the PRs for  merging branches into `dev`).

**Review by popescu-v — CHANGES_REQUESTED** [review]
See comments.

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md (hunk 1 above)
Comment (folmos-at-orange) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:15]: I'll reword to add these points.

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md (hunk 2 above)
Comment (folmos-at-orange) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:2]: Ok. I'll change the title and see if I can put some constraints.

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md (hunk 1 above)
Comment (folmos-at-orange) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: Done

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md (hunk 2 above)
Comment (folmos-at-orange) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: In the end I think it is better just to keep one template: The merges `dev`-> `main` are rare and we know that we can safely ignore most of the template.

If find too much complex to have another template just for these rare merges: The gain is very limited.

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md (hunk 1 above)
Comment (popescu-v) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: Ok.

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md (hunk 2 above)
Comment (popescu-v) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: OK, fair enough. However, I would just one note that all those steps can be sidestepped for these kinds of PRs.

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+---
+
+### TODO Before Asking for a Review
+- [ ] Rebase your branch to the latest version of `dev` (or `main` for release PRs)
+- [ ] Make sure all CI workflows are green
+- [ ] When adding a public feature/fix: Update the `Unreleased` section of `CHANGELOG.md` (no date)
+- [ ] Self-Review: Review "Files Changed" tab and fix any problems you find
+- API Docs (only if there are changes in docstrings, rst files or samples):
+  - [ ] Check the docs build **without** warning
+  - [ ] Check that your changes render well in HTML
```
[the remark this answers is not in the document]
Comment (folmos-at-orange) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:21]: Way too complicated. It is easier to build the docs and check them locally, I can point to the proper doc.

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md (hunk 2 above)
Comment (folmos-at-orange) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: Let's not overload the checklist. I think this is fine.

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (popescu-v) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:21]: IMHO, we need to rigorously check the built artifact: building locally has 2 drawbacks IMHO:
1. We need to set-up and maintain a whole environment for this
2. We have no guarantee that this environment (whence the result of the build) is 100 % in sync with the result of the workflow.

However, I agree that we can just open `index.html` in the browser, there is no need for the HTTP server launch.

[the remark this answers is not in the document]
Comment (folmos-at-orange) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:21]: A compromise after offline discussion: Download the artifact and say to load `index.html` in a browser (usually it is just a double click)

**Review by popescu-v — APPROVED** [review]
LGTM

**Pull request merged**